### PR TITLE
updated get_csid() to handle new ChemSpider API

### PR DESCRIPTION
**Pull request by stitam**
A lot of files seem like they changed, it's because of the devtools::document(). I only worked with the chemspider.R and DESCRIPTION files.

### CODE REVIEW

File: R/chemspider.R [hunk 1]
```diff
@@ -1,66 +1,60 @@
...
+  orderBy_values=c("recordId", "massDefect", "molecularWeight","referenceCount", "dataSourceCount", "pubMedCount", "rscCount")
+  orderDirection_values=c("ascending","descending")
+
+  if(orderBy%in%orderBy_values==FALSE) stop("Invalid argument: orderBy")
+  if(orderDirection%in%orderDirection_values==FALSE) stop("Invalid argument: orderDirection")
+
+  headers <- c("Content-Type" = "","apikey" = apikey)
+  body <-data.frame("name" = query, "orderBy" = orderBy, "orderDirection" = orderDirection)
+  body <- jsonlite::toJSON(body)
+  body <- gsub("\\[|\\]","",body)
```
Comment (EDiLD) [inline R/chemspider.R]: No need for gsub, this should do also the trick? (untested)
```suggestion
body <- list"name" = query, "orderBy" = orderBy, "orderDirection" = orderDirection)
jsonlite::toJSON(body, auto_unbox = TRUE)
```

File: R/chemspider.R [hunk 2]
```diff
@@ -1,66 +1,60 @@
...
-      message('No csid found... Returning NA.')
-      return(NA)
-    }
-    if (first)
-      out <- out[1]
-    names(out) <- NULL
-    return(out)
+get_csid <- function(query, apikey, orderBy="recordId",orderDirection="ascending",  ...){
+
+  orderBy_values=c("recordId", "massDefect", "molecularWeight","referenceCount", "dataSourceCount", "pubMedCount", "rscCount")
```
Comment (EDiLD) [inline R/chemspider.R]: Can you add spaces around all equal signs and try to stick to 80 - 100 character line length (see also https://style.tidyverse.org/).

File: R/chemspider.R [hunk 3]
```diff
@@ -1,66 +1,60 @@
...
+    warning(http_status(postres)$message)
   }
-  out <- lapply(query, foo, token = token, first = first, verbose = verbose)
-  out <- setNames(out, query)
-  if (first)
-    out <- unlist(out)
   return(out)
 }
-
+#test: for a compound of one word, for a compound of multiple words
```
Comment (EDiLD) [inline R/chemspider.R]: Can you update also the unit tests in `tests/testthat/test-chemspider.R` ?

Comment (EDiLD) [inline R/chemspider.R]: Feel free to add yourself as contributor to the `DESCRIPTION` file.

Comment (EDiLD) [inline R/chemspider.R]: Can you also update the `NEWS` file and optionally bump the minor version number?

File: R/chemspider.R (hunk 2 above)
Comment (stitam) [reply R/chemspider.R]: I'd like some help with this, how should I amend my commit? https://www.burntfen.com/2015-10-30/how-to-amend-a-commit-on-a-github-pull-request. Option 2?

Comment (Aariq) [reply R/chemspider.R]: If you just make a new commit in your branch and push it, it will update the pull request I think.

Comment (stitam) [reply R/chemspider.R]: Thank @Aariq I pushed my new commit(s).

File: R/chemspider.R (hunk 3 above)
Comment (stitam) [reply R/chemspider.R]: Added my own apikey and five tests for get_csid().

Comment (stitam) [reply R/chemspider.R]: Thanks:) done.

Comment (stitam) [reply R/chemspider.R]: Done. Both.

**Comment by stitam** [comment]
R CMD check succeeded, I am clueless with this Travis error, any help would be appreciated.

**Comment by EDiLD** [comment]
Don't know what's wrong with travis, but this looks really good!
Merged!
Thanks you for your contribution!

**Pull request merged**